containers/src/Data/IntMap/Internal.hs

```diff
@@ -2578,6 +2579,17 @@ filter :: (a -> Bool) -> IntMap a -> IntMap a
 filter p m
   = filterWithKey (\_ x -> p x) m
 
+-- | \(O(n)\). Filter all keys that satisfy some predicate.
+--
+-- > filterKeys (> 4) (fromList [(5,"a"), (3,"b")]) == singleton 5 "a"
+
+filterKeys :: (Key -> Bool) -> IntMap a -> IntMap a
+filterKeys predicate = go
+    where
+    go Nil           = Nil
+    go t@(Tip k _)   = if predicate k then t else Nil
+    go (Bin p m l r) = bin p m (go l) (go r)
```

Comment:
I agree with @konsumlamm - do we gain anything here from duplicating the definition of `filterWithKey` here instead of just issuing

```haskell
filterKeys p = filterWithKey (\k _ -> p k)
```

As we do in `filter`?

Reply:
@meooow25 suggested implementing it using `filterWithKey`, but I agree, I don't see any reason not to use `filterWithKey` here, especially since `filter` does the same. It's also more obviously correct.

Reply:
@emilypi I imagine that depends on inlining, but we should be able to force the matter if necessary. It's a little better not to call the predicate through a partial application object. That said, I dunno how big a difference it will make.